libmscore/timesig.cpp

```diff
@@ -215,6 +215,17 @@ void TimeSig::read(XmlReader& e)
             _sig.set(z1+z2+z3+z4, n);
             }
       _stretch.reduce();
+
+      // HACK: handle time signatures from scores before 3.5.0 differently on some special occasions.
+      // See https://musescore.org/node/308139.
+      if (score()->mscoreVersion() < QString("3.5.0")) {
```

Comment:
This works but I wonder if we have a more direct way of detecting this. Perhaps we need to consider adding something like a "Protocol version" (concept borrowed from Minecraft) which is an increasing integer and differs for every version to formulate the comparison between minor updates.

Reply:
We don't currently have a better way.  But see the use of this function in lyrics.cpp for a model, I adapted that from some other code that no longer exists.  A few differences worth noting:

- I used masterScore() rather than score()
- I also checked if it was empty
- I omit the last digit from the comparison
- I don't explicitly create a QString

The first two of these are important to get this working right in linked parts, which apparently don't reliably have the field set themselves.  See https://github.com/musescore/MuseScore/commit/30e1ea9674e64511523cf09c35f1c8a45fc92add

Omitting the last digit seems wise because I don't know that we can prove the version would never be just plain 3.5 instead of 3.5.0 (e.g., some nightly or self-builds? the alpha or beta?), and we want to handle that correctly.

Anyhow, this is also the approach I used in #6259, so assuming that is merged as is, good to be consistent.

Reply:
What's the use of checking if it's empty?

Reply:
Also, there's a bigger problem with linked parts: the special texts don't show up at all, or are even saved.

Reply:
The point of checking for empty is to make sure we don't treat such scores as pre-3.5.  I mean, if this string is empty (which it could be for scores saved in test mode or edited by hand, perhaps also for some imports), we don't *know* it's version, but I'd rather see us treat it as "normal".

Reply:
I'm not understanding the issue with parts, it worked fine for me.  The properties aren't linked so if I change the score after generating the parts, I need to change it in both places, but aside from that, everything worked including save/restore.  But, probably you were doing something different from what I tried.